src/server/_query.py

```diff
@@ -466,26 +466,37 @@ def where_time_pair(
         )
         return self
 
+    def apply_lag_filter(self, history_table: str, lag: Optional[int]):
+        if lag is not None:
+            self.retable(history_table)
+            # history_table has full spectrum of lag values to search from whereas the latest_table does not
+            self.where(lag=lag)
+
+    def apply_issues_filter(self, history_table: str, issues: Optional[TimeValues]):
+        if issues:
+            self.retable(history_table)
+            self.where_integers("issue", issues)
+
+    def apply_as_of_filter(self, history_table: str, as_of: Optional[int]):
+        if as_of is not None:
+            self.retable(history_table)
+            sub_condition_asof = "(issue <= :as_of)"
+            self.params["as_of"] = as_of
+            sub_fields = "max(issue) max_issue, time_type, time_value, `source`, `signal`, geo_type, geo_value"
+            sub_group = "time_type, time_value, `source`, `signal`, geo_type, geo_value"
+            sub_condition = f"x.max_issue = {self.alias}.issue AND x.time_type = {self.alias}.time_type AND x.time_value = {self.alias}.time_value AND x.source = {self.alias}.source AND x.signal = {self.alias}.signal AND x.geo_type = {self.alias}.geo_type AND x.geo_value = {self.alias}.geo_value"
+            self.subquery = f"JOIN (SELECT {sub_fields} FROM {self.table} WHERE {self.conditions_clause} AND {sub_condition_asof} GROUP BY {sub_group}) x ON {sub_condition}"
+
     def set_fields(self, *fields: Iterable[str]) -> "QueryBuilder":
         self.fields = [f"{self.alias}.{field}" for field_list in fields for field in field_list]
         return self
 
-    def set_order(self, *args: str, **kwargs: Union[str, bool]) -> "QueryBuilder":
+    def set_sort_order(self, *args: str):
         """
         sets the order for the given fields (as key word arguments), True = ASC, False = DESC
         """
 
-        def to_asc(v: Union[str, bool]) -> str:
-            if v is True:
-                return "ASC"
-            elif v is False:
-                return "DESC"
-            return cast(str, v)
-
-        args_order = [f"{self.alias}.{k} ASC" for k in args]
-        kw_order = [f"{self.alias}.{k} {to_asc(v)}" for k, v in kwargs.items()]
-        self.order = args_order + kw_order
-        return self
+        self.order = [f"{self.alias}.{k} ASC" for k in args]
```

Comment:
This is actually one of the refactors I wanted to apply to `query.py` - the `return self` could be used for method cascading:
```
q.apply_issues_filter().apply_lag_filter().apply_as_of_filter()
```
but it is not used that way anywhere in the codebase, so the statements serve no purpose. Do we still need to keep the `return self` for some other reason? (Or perhaps start using that style of cascading after all?)

Reply:
we can allow method cascading without requiring it.  the rest of the methods of `QueryBuilder` do `return self`, and its best to keep that consistent especially in case someone does eventually choose to write something with cascading.  its also not bad form to return `self` on mutator methods from a functional programming perspective.

Reply:
Got it! I actually wanted to remove `return self` from those other methods too but you make a great point about the functional programming angle.